cartoframes/data/services/bq_datasets.py

```diff
@@ -0,0 +1,50 @@
+import requests
+from carto.utils import ResponseStream
+# from carto.auth import APIKeyAuthClient
+
+from carto.exceptions import CartoException
+
+# TODO: this shouldn't be hardcoded
+DO_ENRICHMENT_API_URL = 'http://localhost:7070/bq'
+
+
+class BQDataset:
+
+    def __init__(self, name_id):
+        self.name = name_id
+        # TODO fix this crap
+        self.session = requests.Session()
+        self.api_key = 'my_valid_api_key'
+
+    def download(self):
+        url = DO_ENRICHMENT_API_URL + '/datasets/' + self.name
+        params = {'api_key': self.api_key}
+
+        try:
+            response = self.session.get(url,
+                                        params=params,
+                                        stream=True)
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            if 400 <= response.status_code < 500:
+                # Client error, provide better reason
+                reason = response.json()['error'][0]
+                error_msg = u'%s Client Error: %s' % (response.status_code,
+                                                      reason)
+                raise CartoException(error_msg)
+            else:
+                raise CartoException(e)
+        except Exception as e:
+            raise CartoException(e)
+
+        return response
+
+    def download_stream(self):
+        return ResponseStream(self.download())
+
+
+class BQUserDataset:
+
+    @staticmethod
+    def name(name_id):
```

Comment:
Right now, in CF we are based on functions instead of classes, but in `carto-python` we are class-based. 

So, in here in CF I would spec something like:
```
from bigquery import read_bigquery

dataFrame = read_bigquery('dataset_name')
``` 

but for `carto-python`, the current one makes sense

Reply:
We talked offline about the python api.

To summarize and for the record: On the one hand, this `function(input) -> output` adapts better to existing recommendations. OTOH, it is not fully possible in all cases without "guessing" the schema.

Let's do the following: let's stick to the proposed python api (private, undocumented), complete a full iteration (putting all the components together to work), then let's consider possible code reorgs or alternate API's.